texar/torch/modules/tokenizers/pretrained_bert_tokenizer.py

```diff
@@ -0,0 +1,199 @@
+        split_tokens = []
+        if self.do_basic_tokenize:
+            for token in self.basic_tokenizer.tokenize(
+                    text, never_split=self.all_special_tokens):
+                assert token is not None
+                for sub_token in self.wordpiece_tokenizer.tokenize(token):
+                    split_tokens.append(sub_token)
+        else:
+            split_tokens = self.wordpiece_tokenizer.tokenize(text)
+        return split_tokens
+
+    def save_vocabulary(self, save_directory: str) -> Tuple[str]:
+        r"""Save the tokenizer vocabulary to a directory or file."""
+        index = 0
+        if os.path.isdir(save_directory):
+            save_directory = os.path.join(save_directory,
+                                          self._VOCAB_FILE_NAMES['vocab_file'])
+        with open(save_directory, "w", encoding="utf-8") as writer:
+            for token, token_index in sorted(self.vocab.items(),
+                                             key=lambda kv: kv[1]):
+                if index != token_index:
+                    print("Saving vocabulary to {}: vocabulary indices are not "
+                          "consecutive. Please check that the vocabulary is "
+                          "not corrupted!".format(save_directory))
+                    index = token_index
+                writer.write(token + u'\n')
+                index += 1
+
+        return (save_directory, )
+
+    @property
+    def vocab_size(self) -> int:
+        return len(self.vocab)
+
+    def _convert_token_to_id(self, token: str) -> int:
+        r"""Converts a token (str/unicode) in an id using the vocab."""
+        unk_id = self.vocab.get(self.unk_token)
+        assert unk_id is not None
+        return self.vocab.get(token, unk_id)
+
+    def _convert_id_to_token(self, index: int) -> str:
+        r"""Converts an index (integer) in a token (string/unicode) using
+        the vocab.
+        """
+        return self.ids_to_tokens.get(index, self.unk_token)
+
+    def convert_tokens_to_string(self, tokens: List[str]) -> str:
+        r"""Converts a sequence of tokens (string) in a single string."""
+        out_string = ' '.join(tokens).replace(' ##', '').strip()
+        return out_string
+
+    @staticmethod
+    def default_hparams() -> Dict[str, Any]:
+        return {
+            'pretrained_model_name': 'bert-base-uncased',
+            'vocab_file': None,
+            'max_len': 512,
+            'unk_token': '[UNK]',
+            'sep_token': '[SEP]',
+            'pad_token': '[PAD]',
+            'cls_token': '[CLS]',
+            'mask_token': '[MASK]',
+            'tokenize_chinese_chars': True,
+            'do_lower_case': True,
+            'do_basic_tokenize': True,
+            'never_split': None,
+            '@no_typecheck': ['pretrained_model_name'],
+        }
+
+    @classmethod
+    def _transform_config(cls, pretrained_model_name: str,
+                          cache_dir: str):
+        return
```

Comment:
Let's use `pass` to indicate no-op. `return` feels like explicitly returning `None`.